### PR TITLE
Added request input validation on API routes

### LINKED ISSUE #484: bug: No input validation on backend api routes
code

right now there is no input validation on backend

we can create a zod schema to validate the input on backend and reuse the same schema in react-hook-forms to validate the form input on frontend. 

I would like to work on this issue.

### PR DESCRIPTION
**Pull request by hxdy-1**
### PR Fixes:
- 1 This PR adds validations for request inputs on the backend API routes wherever it was needed.
- 2 Also fixes some wrong status codes in some routes (440 was being used instead of 404).

Resolves #484  

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

@hkirat Those big differences in files changed is just because zod schemas were created there

### CODE REVIEW

**Comment by hxdy-1** [comment]
@hkirat could you please review this PR :)

File: src/app/api/course/videoProgress/route.ts [hunk 1]
```diff
@@ -26,8 +27,21 @@ export async function GET(req: NextRequest) {
   });
 }
 
+const requestBodySchema = z.object({
+  contentId: z.number(),
+  currentTimestamp: z.number(),
```
Comment (hkirat) [inline src/app/api/course/videoProgress/route.ts:32]: have u tested this?
does the frontend indeeed send a number or a string?

Comment (hxdy-1) [reply src/app/api/course/videoProgress/route.ts:32]: yessir, i have tested

**Pull request merged**